File: iqchannels-sdk/src/main/java/ru/iqchannels/sdk/download/FileDownloader.java
package ru.iqchannels.sdk.download;

import android.app.DownloadManager;
import android.content.Context;
import android.net.Uri;
import android.os.Environment;
import ru.iqchannels.sdk.R;

public class FileDownloader {

    public static void downloadFile(Context context, String fileUrl, String fileName) {
        DownloadManager downloadManager = (DownloadManager) context.getSystemService(Context.DOWNLOAD_SERVICE);

        Uri uri = Uri.parse(fileUrl);

        DownloadManager.Request request = new DownloadManager.Request(uri);
        request.setAllowedNetworkTypes(DownloadManager.Request.NETWORK_WIFI | DownloadManager.Request.NETWORK_MOBILE)
                .setAllowedOverRoaming(false)
                .setTitle(fileName)
                .setDescription(context.getString(R.string.downloading))
                .setDestinationInExternalPublicDir(Environment.DIRECTORY_DOWNLOADS, fileName);

        downloadManager.enqueue(request);
    }
}

File: iqchannels-sdk/src/main/java/ru/iqchannels/sdk/download/FileDownloader.kt
package ru.iqchannels.sdk.download

import android.content.ContentValues
import android.content.Context
import android.os.Build
import android.os.Environment
import android.provider.MediaStore
import androidx.annotation.RequiresApi

object FileDownloader {

	@RequiresApi(Build.VERSION_CODES.Q)
	fun saveFileToDownloads(context: Context?, fileName: String, fileContent: ByteArray) {
		val contentResolver = context?.contentResolver
		val contentValues = ContentValues().apply {
			put(MediaStore.Downloads.DISPLAY_NAME, fileName)
			put(MediaStore.Downloads.RELATIVE_PATH, Environment.DIRECTORY_DOWNLOADS)
		}

		val uri = contentResolver?.insert(MediaStore.Downloads.EXTERNAL_CONTENT_URI, contentValues)
		uri?.let {
			contentResolver.openOutputStream(it)?.use { outputStream ->
				outputStream.write(fileContent)
			}
		}
	}
}